libyang/schema.py

```diff
@@ -394,7 +394,7 @@ def __str__(self):
 class ExtensionParsed(Extension):
     __slots__ = ("module_parent",)
 
-    def __init__(self, context: "libyang.Context", cdata, module_parent: Module = None):
+    def __init__(self, context: "libyang.Context", cdata, module_parent: Module):
```

Comment:
Is this related to the extension plugin?

Reply:
Yes, it turned out that extension parsed must have always module defined. Otherwise the rest of the code to get to `parent_node()` or `module_from_parsed()` doesn't really work

Reply:
Two remarks:
- that's an api breakage, so it is not too nice
- this commit instantiates  an ExtensionParsed object with a None module in libyang_c_lyplg_ext_parse_free_clb, pointing out that there may be a need for it

So I would rather have _module_from_parsed take care of the case where the ExtensionParsed instance has no module, as we do in other parts of the code for cdata or cdata_parsed. Raising self.context.error("cannot get module") if self.module_parent is not set in module_from_parsed would do the trick, right?

Reply:
Fixed